### PR TITLE
Base container

### PR DESCRIPTION
**Pull request by drivera-armedia**
First version

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1 +1,185 @@
...
+ARG PKG="google_cadvisor"
+ARG SRC="https://github.com/google/cadvisor.git"
+
+#
+# Some important labels
+#
+LABEL ORG="Armedia LLC"
+LABEL MAINTAINER="Armedia Devops Team <devops@armedia.com>"
+LABEL APP="Google cAdvisor"
+LABEL VERSION="${VER}"
```
Comment (fabricetriboix) [inline Dockerfile:24]: I just realized I don't think we need to add labels for intermediate images. What do you think?

File: Dockerfile [hunk 2]
```diff
@@ -1 +1,185 @@
...
+# Download and install go
+#
+RUN curl -L "${GO_SRC}" -o - | tar -C "/usr/local" -xzf -
+
+#
+# Some important labels
+#
+LABEL ORG="Armedia LLC"
+LABEL MAINTAINER="Armedia Devops Team <devops@armedia.com>"
+LABEL APP="Grafana"
```
Comment (fabricetriboix) [inline Dockerfile]: Grafana?

File: Dockerfile [hunk 3]
```diff
@@ -1 +1,185 @@
...
+
+#
+# Some useful environment variables
+#
+ENV GOROOT="/usr/local/go"
+ENV GOPATH="/go"
+ENV CADVISOR_HEALTHCHECK_URL="http://localhost:8080/healthz"
+
+#
+# Missing dependencies from Alpine:
```
Comment (fabricetriboix) [inline Dockerfile]: I guess this comment is irrelevant since we are using CentOS, not Alpine

File: Dockerfile [hunk 4]
```diff
@@ -1 +1,185 @@
...
+#
+COPY --from=build /usr/local/lib/libpfm.so* /usr/local/lib/
+# COPY --from=build /usr/local/lib/libipmctl.so* /usr/local/lib/
+COPY --from=build "${GOPATH}/src/github.com/google/cadvisor/cadvisor" /usr/local/bin/cadvisor
+
+#
+# Final parameters
+#
+USER        ${UID}
+EXPOSE      8080
```
Comment (fabricetriboix) [inline Dockerfile:185]: Is the port number fixed? Do we want to make it a parameter to this Docker image?

**Review by fabricetriboix — DISMISSED** [review]

File: Dockerfile (hunk 1 above)
Comment (drivera-armedia) [reply Dockerfile:24]: I thought of that repeatedly on the Grafana one that has a bunch ... then I figured I'd just leave them in for consistency's sake, _just in case_.

They do no harm, and adding them helps for form a solid habit of maintaining them...

File: Dockerfile (hunk 2 above)
Comment (drivera-armedia) [reply Dockerfile]: I thought labels didn't matter in intermediates :D

J/k.... I fixed it :)

File: Dockerfile (hunk 3 above)
Comment (drivera-armedia) [reply Dockerfile]: Yes it was a note for me to remember things that were in the Alpine version that didn't make it into the CentOS 7 version. I'd still like to keep the comment b/c they're potentially useful things for the future.

File: Dockerfile (hunk 4 above)
Comment (drivera-armedia) [reply Dockerfile:185]: EXPOSE is a static parameter that can't be changed at runtime ... if you want to remap it, it should be remapped at the upper level using the standard port mapping techniques.

File: Dockerfile (hunk 1 above)
Comment (fabricetriboix) [reply Dockerfile:24]: OK

File: Dockerfile (hunk 3 above)
Comment (fabricetriboix) [reply Dockerfile]: I think that's confusing. Maybe make the comment more explicit, like "This is a note just in case we decide to use Alpine"?

File: Dockerfile (hunk 4 above)
Comment (fabricetriboix) [reply Dockerfile:185]: I know. I am just asking whether the port is configurable or whether it is hardcoded? This question is not related to the EXPOSE command, sorry for the confusion.

Comment (drivera-armedia) [reply Dockerfile:185]: It's configurable via CLI flag

File: Dockerfile (hunk 3 above)
Comment (drivera-armedia) [reply Dockerfile]: Yes, I can do something like that.

File: Dockerfile (hunk 4 above)
Comment (fabricetriboix) [reply Dockerfile:185]: I don't understand your answer...

Furthermore, I can see the `CADVISOR_HEALTHCHECK_URL` has port 8080 hardcoded in it, so as it stands it is not possible to change the port, whether at build time nor at run time.

Comment (fabricetriboix) [reply Dockerfile:185]: Anyway, I just wanted to understand whether the port was configurable or not, and if not whether we want to make so or not. No big deal.

**Review by fabricetriboix — DISMISSED** [review]
